src/System.ComponentModel.TypeConverter/src/System/ComponentModel/BaseNumberConverter.cs

```diff
@@ -111,13 +115,18 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
             {
                 return Convert.ChangeType(value, destinationType, culture);
             }
+
             return base.ConvertTo(context, culture, value, destinationType);
         }
 
         public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
         {
-            return base.CanConvertTo(context, destinationType) || destinationType.IsPrimitive;
+            if (destinationType != null && destinationType.IsPrimitive)
+            {
+                return true;
+            }
+
+            return base.CanConvertTo(context, destinationType);
```

Comment:
Why did you change this?

Reply:
This fixes a `NullReferenceException` if `destinationType` is null